app/scripts/controllers/transactions/index.js

```diff
@@ -331,6 +331,195 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
+  // ====================================================================================================================================================
+
+  /**
+   * @param {number} txId
+   * @returns {TransactionMeta} the txMeta who matches the given id if none found
+   * for the network returns undefined
+   */
+  _getTransaction(txId) {
+    const { transactions } = this.store.getState();
+    return transactions[txId];
+  }
+
+  _checkIfTxStatusIsUnapproved(txId) {
+    return (
+      this._getTransaction(txId).status === TRANSACTION_STATUSES.UNAPPROVED
+    );
+  }
+
+  _updateTransaction(txId, txParams, note) {
+    const txMeta = this._getTransaction(txId);
+    const updatedMeta = { ...txMeta, ...txParams };
+    this.txStateManager.updateTransaction(updatedMeta, note);
+  }
+
+  /**
+   * updates the gas fees of the transaction with id if the transaction state is unapproved
+   *
+   * @param {string} txId - transaction id
+   * @param {object} txGasFees - holds the gas fees parameters
+   * {
+   * gasLimit,
+   * gasPrice,
+   * maxPriorityFeePerGas,
+   * maxFeePerGas,
+   * estimateUsed
+   * }
+   * @param txGasFees.gasLimit
+   * @param txGasFees.gasPrice
+   * @param txGasFees.maxPriorityFeePerGas
+   * @param txGasFees.maxFeePerGas
+   * @param txGasFees.estimateUsed
+   */
+  updateTransactionGasFees(
+    txId,
+    { gasLimit, gasPrice, maxPriorityFeePerGas, maxFeePerGas, estimateUsed },
+  ) {
+    if (!this._validateTransaction(txId)) {
+      return;
+    }
+
+    const txGasFees = {
+      txParams: {
```

Comment:
In this case, we need to be careful not to overwrite the other properties of `txParams`. So we should probably add the existing txParams properties within `txGasFees`. Something like:
```
const { txParams: currentTxParams, estimateUsed: currentEstimateUsed } = this._getTransaction(txId);
    const txGasFees = {
        txParams: {
            ...currentTxParams
            gasLimit: gasLimit || currentTxParams.gasLimit,
            gasPrice: gasPrice || currentTxParams.gasPrice,
            maxPriorityFeePerGas: maxPriorityFeePerGas || currentTxParams.maxPriorityFeePerGas,
            maxFeePerGas: maxFeePerGas || currentTxParams.maxFeePerGas,
          },
          estimateUsed: estimateUsed || currentEstimateUsed,
    };
```

Reply:
The test cases caught this ;) 
I used lodash `merge` instead. One liner and cleaner.

```
    const txMeta = this.txStateManager.getTransaction(txId);
    const updated = merge(txMeta, proposedUpdate);
    this.txStateManager.updateTransaction(updated, note);
```